### PR TITLE
fix: Add DeleteRolesForUserInDomain

### PR DESCRIPTION
**Pull request by dovics**
fix: https://github.com/casbin/casbin/issues/522
Signed-off-by: dovics <wrs369@163.com>

### CODE REVIEW

**Comment by hsluoyz** [comment]
@95Key can you review this PR?

File: rbac_api_with_domains.go [hunk 1]
```diff
@@ -42,3 +42,19 @@ func (e *Enforcer) AddRoleForUserInDomain(user string, role string, domain strin
 func (e *Enforcer) DeleteRoleForUserInDomain(user string, role string, domain string) (bool, error) {
 	return e.RemoveGroupingPolicy(user, role, domain)
 }
+
+// DeleteRolesForUserInDomain deletes all roles for a user inside a domain.
+// Returns false if the user does not have any roles (aka not affected).
+func (e *Enforcer) DeleteRolesForUserInDomain(user string, domain string) (bool, error) {
```
Comment (nodece) [inline rbac_api_with_domains.go:48]: Using `e.RemoveFilteredGroupingPolicy` is easier.

Comment (dovics) [reply rbac_api_with_domains.go:48]: `e.RemoveFilteredGroupingPolicy` can only filter one field, how to find `domain` and `user` at the same time?

Comment (nodece) [reply rbac_api_with_domains.go:48]: okay, you are right.

**Pull request merged**